### DISCUSSION TITLE
Enhancing Autocomplete Text Input with Dynamic API Calls and Option Updates

### DISCUSSION THREAD

**Discussion by SagarPanchasara** [srivastavaanurag79/react-native-paper-select#39] [General]
Hi Dev,

I'm exploring ways to trigger API calls on text change event in the search box, aiming to dynamically update the options array list. The objective is to integrate autocomplete functionality into the user interface (to give the same UI as the 'react-native-paper-select' component).

Are there any alternative approaches to achieve this?

Your insights would be greatly appreciated.

Regards,
Sagar Panchasara

**Comment by srivastavaanurag79** [comment]
Yes, it can be implemented, will need to add a onChange function in the search bar with min three characters which will triger the API call and update the array list and on clear button click will call the initial list api. I will try adding it in the new build.
